### ISSUE TITLE
Math block: Syntax error notifications not accessible.

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#73445]
### Description

If a user enters invalid Latex notations, an error is shown visibly. This error is not explicitly connected to the `textarea` and is not announced to screen readers.

### Step-by-step reproduction instructions

1. Enable screen reader.
2. Add math block.
3. Type x&

You should see a visual warning. It will not be spoken and the user will not have any equivalent feedback.

### Screenshots, screen recording, code snippet

<img width="417" height="270" alt="Image" src="https://github.com/user-attachments/assets/299969e3-09a8-497c-b452-fd1a785beb74" />

### Environment info

- WP 6.9-RC2-61266, Gutenberg 22.1.0
- Chrome/NVDA

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
